### PR TITLE
upgrade hive

### PR DESCRIPTION
**Pull request by ldpliu**
Signed-off-by: ldpliu <daliu@redhat.com>
https://github.com/stolostron/backlog/issues/22810

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -71,63 +72,66 @@ require (
...
 replace (
-	k8s.io/api => k8s.io/api v0.23.3
-	k8s.io/apimachinery => k8s.io/apimachinery v0.23.3
-	k8s.io/apiserver => k8s.io/apiserver v0.23.3
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.23.3
-	k8s.io/client-go => k8s.io/client-go v0.23.3
-	k8s.io/code-generator => k8s.io/code-generator v0.23.3
-	k8s.io/component-base => k8s.io/component-base v0.23.3
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.23.3
+	github.com/openshift/api => github.com/openshift/api v0.0.0-20220531073726-6c4f186339a7
```
Comment (skeeey) [inline go.mod:127]: why not use these directly?

File: pkg/helpers/helpers.go [hunk 2]
```diff
@@ -372,31 +372,25 @@ func ApplyResources(clientHolder *ClientHolder, recorder events.Recorder,
...
 		case *appsv1.Deployment:
 			errs = append(errs, applyDeployment(clientHolder, recorder, required))
 		case *rbacv1.ClusterRole:
-			_, _, err := resourceapply.ApplyClusterRole(clientHolder.KubeClient.RbacV1(), recorder, required)
+			_, _, err := resourceapply.ApplyClusterRole(context.TODO(), clientHolder.KubeClient.RbacV1(), recorder, required)
 			errs = append(errs, err)
 		case *rbacv1.ClusterRoleBinding:
-			_, _, err := resourceapply.ApplyClusterRoleBinding(clientHolder.KubeClient.RbacV1(), recorder, required)
-			errs = append(errs, err)
-		case *crdv1beta1.CustomResourceDefinition:
```
Comment (skeeey) [inline pkg/helpers/helpers.go:391]: we need this for ocp 311, as a temporary way, you can copy the old code to here

Comment (ldpliu) [reply pkg/helpers/helpers.go:391]: changed, please reiview it again

File: go.mod (hunk 1 above)
Comment (ldpliu) [reply go.mod:127]: done

**Pull request merged**